sphinx/cmdline.py

```diff
@@ -83,6 +84,19 @@ def handle_exception(app, args, exception, stderr=sys.stderr):
                   file=stderr)
 
 
+def jobs_argument(value):
+    # type (int, str) -> int
+    """
+    Special type to handle 'auto' flags passed to 'sphinx-build' via -j flag. Can
+    be expanded to handle other special scaling requests, such as setting job count
+    to (cpu_count / 2)
+    """
+    if value == 'auto':
+        return multiprocessing.cpu_count()
+    else:
+        return value
```

Comment:
I _think_ you should be checking that this is a int, e.g. `return int(value)`

Reply:
Yep - otherwise, it's still a string passed from the command line.